Source/WebCore/style/StyleTreeResolver.cpp

```diff
@@ -504,11 +495,13 @@ ResolutionContext TreeResolver::makeResolutionContext()
 
 ResolutionContext TreeResolver::makeResolutionContextForPseudoElement(const ElementUpdate& elementUpdate, PseudoId pseudoId)
 {
-    auto parentStyle = [&] {
+    auto parentStyle = [&]() -> const RenderStyle* {
         if (pseudoId == PseudoId::FirstLetter) {
             if (auto* firstLineStyle = elementUpdate.style->getCachedPseudoStyle(PseudoId::FirstLine))
                 return firstLineStyle;
         }
+        if (pseudoId == PseudoId::Backdrop)
+            return &m_parentStack.first().style;
```

Comment:
I don't understand this change. It is not mentioned in the commit message either. Why is it needed? Where is it tested?